File: programmers/dfs_and_bfs/PR_43165_타겟넘버.py
import sys
sys.setrecursionlimit(10**6)

def solution(numbers, target):
    result = []

    def dfs(count, total):
        if count == len(numbers):
            print(total)
            if total == target:
                result.append(total)
            return
        
        dfs(count+1, total + numbers[count])
        dfs(count+1, total - numbers[count])

        return

    dfs(0, 0)

    answer = len(result)
    return answer

numbers = [1,1,1,1,1]
target = 3
print(solution(numbers, target))

File: problems/dfs_and_bfs/PR_43165_타겟넘버.js
function solution(numbers, target) {
  const need_visited = [[0, 0]];
  let total_count = 0;
  
  function dfs(number, count) {
      if (count < numbers.length) {
        dfs(number + numbers[count], count + 1);
        dfs(number - numbers[count], count + 1);
      } else {
        if (number === target) total_count += 1;
      }
  }
  
  dfs(0,0)

  return total_count;
}